### ISSUE TITLE
Help with deploymet 

### ISSUE THREAD

**Issue by andrusstrockiy** [Crapworks/RESTlos#10]
Could you advise on installation to production.

> To use this api with an apache web server, you need to install mod-wsgi.

Done

<pre>httpd -M | grep wsgi
Syntax OK
 wsgi_module (shared)
</pre>

> In the contrib directory you will find a sample apache configuration which can be placed in the conf.d directory.

Done. Also copied all files to 
/opt/restols
as mentioned in /opt/restlos/contrib/restlos-directory.
Then on next step.

> You will also find a file called application.py, which have to be placed in the same directory as the executable python script.This will load the api as a wsgi application into the web server.

Err, didn't found it.

[andruss@serv restlos]# find /opt/restlos/ -name "application" 

Then i head to to http://nagios.domain.net/restlos

And got 503 error

Inside apache error log 

<pre>
[Tue Aug 16 15:46:48 2016] [error] [client 192.168.1.222] (13)Permission denied: mod_wsgi (pid=19069): Unable to connect to WSGI daemon process 'restlos' on '/etc/httpd/logs/wsgi.19065.0.1.sock' after multiple attempts.
</pre>

ls -lh /etc/httpd/logs/

<pre>
srwx------ 1 apache root    0 Aug 16 16:06 wsgi.21577.0.1.sock
</pre>

i.e is avalible '/etc/httpd/logs/wsgi.19065.0.1.sock' but it 0 bytes.

What permissions should be for /etc/httpd/logs/wsgi.19065.0.1.sock and should be restlos running in background ?

**Comment by Crapworks** [comment]
Hi @andrusstrockiy, 

just a quick guess:
Looks like you are running RedHat/CentOS. Can you try to disable SELinux (if it is enabled) and see if that changes something?

Have you edited the apache config file? Which one are you using? Can you give the `ls -lha /opt/restlos` output?

Regards,
Christian

**Comment by Crapworks** [comment]
Also: If you just copied the apache vhost file, you need to at least adjust this line:

`WSGIDaemonProcess restlos user=www-data group=www-data threads=5`

This will work on Debian like systems, since apache is running as ww-data there. You need to change it to the operatingsystem specific user (Seems to be apache, not sure about the group)

**Comment by andrusstrockiy** [comment]
Yes for centos user name is apache and  my line looks as following:

<pre>
WSGIDaemonProcess restlos user=apache group=apache threads=5
</pre>

But i get the same error 503 
also tried to change group  group=root i.e

<pre>
WSGIDaemonProcess restlos user=apache group=root threads=5
</pre>

but again same response 503.
Maybe there is another configuration i need to check and adjust.

FYI: To be able run restlosapi while working on that issue with apache added host parameter to  restlosapi.py file

<pre>
    app.run(port=config['port'],host='0.0.0.0')
</pre>

Hope you don't mind ;-) (and yes i know it's not for production and single threaded)

**Comment by Crapworks** [comment]
So have you tried disabling SELinux? Because for me that looks like an apache/operating system error and nothing related to RESTlos, since you've done exactly what is needed to install it. 

Again, please show me you apache vhost config and the contents of `/opt/restlos`

**Comment by andrusstrockiy** [comment]
> So have you tried disabling SELinux?

Yes of course 

<pre>
[root@server ~]# cat /etc/selinux/config | grep SELINUX
# SELINUX= can take one of these three values:
SELINUX=disabled
</pre>

> please show me you apache vhost

Here is my vhost
cat /etc/httpd/conf.d/restols.conf 

<pre>
# Place this file in /etc/apache/conf.d and restart the webserver.
# For this to work, contrib/restlos.wsgi has to be placed in the toplevel
# directory. In this case, the restlos repository should be placed in 
# /opt/restlos. Change the directories if you decide to put it somewhere else.
WSGIScriptAlias /restlos /opt/restlos/restlos.wsgi
WSGIDaemonProcess restlos user=apache group=apache threads=5
## If you want to use the verify ability, all nagios config files should 
## be accessible from the user the api runs with. In case you dont want to
## use sudo to grant the specific rights, you can run the wsgi application
## as the nagios user. In this case, uncomment the following line:
#WSGIDaemonProcess restlos user=nagios group=nagios threads=5
<Directory /opt/restlos>
    WSGIProcessGroup restlos
    WSGIApplicationGroup %{GLOBAL}
    WSGIPassAuthorization On
    Order deny,allow
    Allow from all
</Directory>
</pre>

> and the contents of /opt/restlos

Here is the listing

[root@server ~]# ls -lh /opt/restlos

<pre>
total 65M
-rw-r--r--  1 apache root  722 Авг 12 16:54 ChangeLog
-rw-r--r--  1 apache root 1022 Авг 12 17:04 config.json
drwxr-xr-x  2 apache root 4,0K Авг 16 15:21 contrib
drwxr-xr-x  2 apache root 4,0K Авг 12 17:04 examples
drwxr-xr-x 18 apache root 4,0K Авг 12 17:54 Python-2.7.8
-rw-r--r--  1 apache root  64M Июн 30  2014 Python-2.7.8.tar
-rw-r--r--  1 apache root 7,9K Авг 12 16:54 README.md
-rwxr-xr-x  1 apache root  14K Авг 16 16:27 restlosapi.py
-rw-r--r--  1 apache root  258 Авг 16 15:27 restlos.wsgi
drwxrwxrwx  9 apache root 4,0K Авг 12 17:56 setuptools-1.4.2
-rw-r--r--  1 apache root 779K Янв  9  2016 setuptools-1.4.2.tar.gz
drwxr-xr-x  2 apache root 4,0K Авг 12 16:54 static
drwxr-xr-x  2 apache root 4,0K Авг 12 16:54 templates
drwxr-xr-x  4 apache root 4,0K Авг 12 17:59 utils
</pre>

**Comment by Crapworks** [comment]
Hi @andrusstrockiy 

did you have any progress here? I just had the chance to test you configuration on a centos system and it works perfectly here. The only thing I had to change, dependent on the apache version

```
        WSGIApplicationGroup %{GLOBAL}
        Require all granted <--
```

```
# httpd -v
Server version: Apache/2.4.6 (CentOS)
Server built:   Jul 18 2016 15:30:14
```

**Comment by andrusstrockiy** [comment]
> did you have any progress here?

Nope.Still keep running like separate process i.e:
`python2.7 /opt/restlos/restlosapi.py 5000 &`

> I just had the chance to test you configuration on a centos system 

Could post small how-to  what you did so i can reproduce on my Centos6 Environment.
